packages/core/test/spawn.test.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  PromiseActorLogic,
+  createActor,
+  createMachine,
+  fromPromise,
+  spawn
+} from '../src';
+
+describe('spawn action', () => {
+  it('can spawn', () => {
+    const actor = createActor(
+      createMachine({
+        entry: spawn(
+          fromPromise(() => Promise.resolve(42)),
+          { id: 'child' }
+        )
+      })
+    );
+
+    actor.start();
+
+    expect(actor.getSnapshot().children.child).toBeDefined();
+  });
+
+  it('can spawn from named actor', () => {
+    const fetchNum = fromPromise(({ input }: { input: number }) =>
+      Promise.resolve(input * 2)
+    );
+    const actor = createActor(
+      createMachine({
+        types: {
+          actors: {} as {
+            src: 'fetchNum';
+            logic: typeof fetchNum;
+          }
+        },
+        entry: spawn('fetchNum', { id: 'child', input: 21 })
```

Comment:
I wonder if we could make it easier to spawn N actors at the same time. I was thinking about the API that would allow for a resolver to return many actors:
```ts
spawn(spawn => ({
  id1: spawn('foo', { input: 42 }),
  id2: spawn('bar', { input: 'answer' })
}))
```

Of course, I don't like using `spawn` name for both here. A nice thing about this one is that we could more easily **derive** spawned IDs in the future from types. Although, perhaps we could also do that from the `spawn` action since it returns, well, an action object - and not the actual actor. So we still have a place to keep the inferred ID at the type level. 

So basically... I'm probably just thinking about something that would allow us to spawn multiple actors at the same time without using `pure`

Reply:
Discussed offline, we'll stick with spawning one at a time (there's `pure` for dynamic spawns)